scripts/seed_users.js

```diff
@@ -0,0 +1,516 @@
+
+    // Initialize common module with settings
+    common.init(settings);
+});
+
+// Define user with parameterized constructor
+class User {
+    constructor(name, type, language, color, password, classroom) {
+        this.name = name;
+        this.type = type;
+        this.language = language;
+        this.color = color;
+        this.password = password;
+        if (classroom)
+            this.classroom = classroom;
+        this.status = 0;
+        this.comment = "";
+        this._id = "";
+    }
+}
+
+// Initialize the array that will contains all the users read from CSV
+var Users = [];
+
+// Trim and lowercase string
+function cleanString(string) {
+    if (!string) return;
+    string = string.trim();
+    string = string.toLowerCase();
+    return string;
+}
+
+// Get random color from xocolors and generate color string
+function getRandomColorString() {
+    var randomColor = xocolors[Math.floor(Math.random()*xocolors.length)];
+    randomColor = JSON.stringify(randomColor);
+    return randomColor;
+}
+
+// Validate color string
+function isValidColor(color) {
+    // Check if valid JSON
+    try {
+        color = JSON.parse(color);
+    } catch (e) {
+        return false;
+    }
+
+    // Check if property 'stroke' and 'fill' exists
+    if (!color.stroke || !color.fill) return false;
+
+    // Look for the color in xocolors
+    for (var i=0; i<xocolors.length; i++) {
+        if (xocolors[i].stroke == color.stroke && xocolors[i].fill == color.fill) {
+            return true;
+        }
+    }
+    return false
+}
+
+// Validate language
+function isValidLanguage(lang) {
+    var sugarizerLang = ["en", "es", "fr", "de", "pt", "ar", "ja", "pl", "ibo", "yor"];
+    if(sugarizerLang.indexOf(lang) == -1) return false;
+    else return true;
+}
+
+// Generate random password
+function generatePassword(length) {
+    var password = "";
+    var possible = "ABCDEFGHIJKLMNOPQRSTUVWXabcdefghijklmnopqrstuvwxyz0123456789"; // All alphanumeric characters except 'Y' and 'Z' 
+    
+    for (var i = 0; i < length; i++)
+        password += possible.charAt(Math.floor(Math.random() * possible.length));
+    
+    return password;
+}
+
+// Validate data row
+function validateUserRow(user, index) {
+    if (!user.name) {
+        console.log("Row: " + (index + 1) + " -> Username is invalid. Skipping...");
+        return;
+    }
+
+    user.type = cleanString(user.type);
+    if (!user.type) {
+        user.type = "student";
+    }
+
+    user.language = cleanString(user.language);
+    if (!isValidLanguage(user.language)) {
+        user.language = "en";
+    }
+
+    if (!isValidColor(user.color)) {
+        user.color = getRandomColorString();
+    }
+
+    var minPass = (settings && settings.security && parseInt(settings.security.min_password_size)) ? parseInt(settings.security.min_password_size) : 4;
+    if (!user.password || typeof user.password != "string" || user.password.length < minPass) {
```

Comment:
You should validate the password string too. You could use the recently merged function to do that: https://github.com/llaske/sugarizer-server/pull/117/files#diff-d95c18556dca4520ddbc77a44ecc3363

Reply:
Rebased with upstream/dev and integrated validation:
https://github.com/llaske/sugarizer-server/blob/322c84b89518722eee33513759a5b13bdea94839/scripts/seed_users.js#L34-L36
https://github.com/llaske/sugarizer-server/blob/322c84b89518722eee33513759a5b13bdea94839/scripts/seed_users.js#L140
https://github.com/llaske/sugarizer-server/blob/322c84b89518722eee33513759a5b13bdea94839/scripts/seed_users.js#L160